src/combinatorics/adj_matrix.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author:  Aaron Anderson, Jalex Stark.
+-/
+import linear_algebra.matrix
+import data.rel
+import combinatorics.simple_graph
+
+open_locale big_operators matrix
+open finset matrix simple_graph
+
+universes u v
+variables {α : Type u} [fintype α]
+variables (R : Type v) [semiring R]
+
+namespace simple_graph
+
+variables (G : simple_graph α) (R) [decidable_rel G.adj]
+
+/-- `adj_matrix G R` is the matrix `A` such that `A i j = (1 : R)` if `i` and `j` are
+  adjacent in the simple graph `G`, and otherwise `A i j = 0`. -/
+def adj_matrix : matrix α α R
+| i j := if (G.adj i j) then 1 else 0
+
+variable {R}
+
+theorem transpose_adj_matrix : (G.adj_matrix R)ᵀ = G.adj_matrix R :=
+by { ext, simp [adj_matrix, edge_symm] }
+
+@[simp]
+lemma adj_matrix_apply (v w : α) : G.adj_matrix R v w = if (G.adj v w) then 1 else 0 := rfl
```

Comment:
If you put this simp lemma before the transpose lemma, the proof of the transpose lemma needs one less argument to simp.